extensions/amp-script/0.1/amp-script.js

```diff
@@ -25,18 +31,33 @@ import {
 import {dev, user} from '../../../src/log';
 import {getMode} from '../../../src/mode';
 import {isExperimentOn} from '../../../src/experiments';
-import {
-  sanitizer,
-  upgrade,
-} from '@ampproject/worker-dom/dist/unminified.index.safe.mjs.patched';
 
 /** @const {string} */
 const TAG = 'amp-script';
 
 /** @const {number} */
 const MAX_SCRIPT_SIZE = 150000;
 
+const PHASE_MUTATING = 2;
+
 export class AmpScript extends AMP.BaseElement {
+
+  /**
+   * @param {!Element} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private @const {!../../../src/service/vsync-impl.Vsync} */
+    this.vsync_ = Services.vsyncFor(this.win);
+
+    /** @private {?WorkerDom} */
+    this.workerDom_ = null;
```

Comment:
Minor: Outside scope of this PR, but should we move other exports into this interface? Or can we export the `terminate` function directly (similar to `upgrade`)? Fine with either one but consistency would be nice.

Reply:
I think `terminate` and `upgrade` are very different. `upgrade`, `install`, `fetch` and such - are all factory methods - they are proxy of `WorkerDom` constructor, albeit asynchronous. `terminate` is the method of the `WorkerDom` once it's created. I could find a way to make `terminate` an exported method, but there'll be some hustle: I'd have to set `WorkerDom` instance as a grafted property on the `baseElement` and `terminate` method would first lookup `WorkerDom` instance and then call the `terminate` method. E.g.

```
export function terminate(baseElement: Element) {
  if (baseElement['WORKER_DOM']) {
    (baseElement['WORKER_DOM'] as WorkerDom).terminate();
  }
}
```

I don't think this would be an improvement - just race conditions along here would make things even worse. This would be especially problematic if number of lifecycle methods in `WorkerDom` would grow.